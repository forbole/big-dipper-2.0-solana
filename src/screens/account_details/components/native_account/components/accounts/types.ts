export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
