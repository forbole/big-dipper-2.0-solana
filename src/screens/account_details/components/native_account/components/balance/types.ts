export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
