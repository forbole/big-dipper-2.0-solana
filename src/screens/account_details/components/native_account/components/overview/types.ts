export type OverviewType = {
  address: string;
}
