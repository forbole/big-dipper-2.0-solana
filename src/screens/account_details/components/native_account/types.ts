export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
