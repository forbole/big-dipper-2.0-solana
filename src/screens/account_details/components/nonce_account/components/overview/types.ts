export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
