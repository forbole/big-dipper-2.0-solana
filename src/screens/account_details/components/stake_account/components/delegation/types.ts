export type DelegationType = {
  activated: TokenUnit;
  activating: TokenUnit;
  total: TokenUnit;
}
