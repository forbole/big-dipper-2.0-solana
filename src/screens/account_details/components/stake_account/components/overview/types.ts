export type OverviewType = {
  address: string;
  balance: number;
  stakeAuthority: string;
  withdrawAuthority: string;
  rentReserve: number;
  validator: string;
  delegated: number;
  rewards: number;
}
