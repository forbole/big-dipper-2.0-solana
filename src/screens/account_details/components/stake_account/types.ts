export type OverviewType = {
  address: string;
  balance: TokenUnit;
  staker: string;
  withdrawer: string;
}

export type StakeAccountState = {
  loading: boolean;
  overview: OverviewType;
}
