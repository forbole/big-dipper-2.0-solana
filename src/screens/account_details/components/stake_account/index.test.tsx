/* eslint-disable */
import React from 'react';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import { StakeAccountDetailsDocument } from '@graphql/types';
import StakeAccount from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      address: "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
    },
  }),
}));

jest.mock('./components', () => ({
  Overview: (props) => <div id="Overview" {...props} />,
  StakeInfo: (props) => <div id="StakeInfo" {...props} />,
}));

const mockData = jest.fn().mockResolvedValue({
  "data": {
    "block": [
      {
        "slot": 131479854
      }
    ],
    "stakeAccount": [
      {
        "address": "Gs2tWRu1Y5FbDmwss74CDcJ2azEsc9RbqZPVMvVbg1bq",
        "nativeBalance": {
          "balance": 430282877615
        },
        "staker": "12Euh1pmCycRLq4grkyY7VDubawWJKXxbX9FJQAs4hZy",
        "withdrawer": "EhYXq3ANp5nAerUpbSgd7VK2RRcxK1zNuSQ755G5Mtxx",
        "stakeDelegation": {
          "stake": 430276028975,
          "activationEpoch": 200,
          "deactivationEpoch": 18446744073709552000,
          "voter": "964w4qykexipZ7aCur1BEeJtexTMa1ehMUc9tCcxm9J3"
        }
      }
    ]
  }
});

// ==================================
// unit tests
// ==================================
describe('screen: StakeAccount', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      StakeAccountDetailsDocument,
      mockData,
    );

    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <StakeAccount />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
