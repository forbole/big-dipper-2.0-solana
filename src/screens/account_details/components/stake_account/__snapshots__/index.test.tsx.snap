// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: StakeAccount matches snapshot 1`] = `
Array [
  <div
    className="makeStyles-overview"
    id="Overview"
    overview={
      Object {
        "address": "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
        "balance": Object {
          "baseDenom": "udaric",
          "displayDenom": "daric",
          "exponent": 6,
          "value": "430282.877615",
        },
        "staker": "12Euh1pmCycRLq4grkyY7VDubawWJKXxbX9FJQAs4hZy",
        "withdrawer": "EhYXq3ANp5nAerUpbSgd7VK2RRcxK1zNuSQ755G5Mtxx",
      }
    }
  />,
  <div
    className="makeStyles-overview"
    id="StakeInfo"
    stake={
      Object {
        "activationEpoch": 200,
        "deactivationEpoch": 18446744073709552000,
        "delegated": Object {
          "baseDenom": "udaric",
          "displayDenom": "daric",
          "exponent": 6,
          "value": "430276.028975",
        },
        "status": "active",
        "voter": "964w4qykexipZ7aCur1BEeJtexTMa1ehMUc9tCcxm9J3",
      }
    }
  />,
]
`;
